### PR TITLE
grpc: do not apply default service config upon receipt of invalid service config

### PR DESCRIPTION
**Pull request by easwars**
In #5238, we made a change to apply the default service config when grpc receives invalid service config from the name resolver (and a default service config exists).

But this does not sit well with [4.2](https://github.com/grpc/proposal/blob/master/A21-service-config-error-handling.md#behavior-on-receiving-a-new-grpc-config):

```
If the client has not previously selected a valid service config or no service config,
for example a new client, then it treats the resolution attempt as having failed
(e.g., for a polling-based resolver, it should retry the query after appropriate backoff).
In other words, the channel will remain in state TRANSIENT_FAILURE until the
resolver returns a valid response.
```

This PR does the following:
- undoes some of the changes from #5238, by not applying the default service config
- Renames `test/resolver_test.go` to `test/config_selector_test.go`  (since it contained only config selector related tests)
- Adds a new `test/resolver_test.go` which contains tests for different error handling scenarios upon receipt of resolver update. This pulls in a couple of tests from `resolver_conn_wrapper_test.go`, which is now deleted.

RELEASE NOTES: none

### CODE REVIEW

**Review by dfawley — APPROVED** [review]
Nice, thanks for adding the new test cases.  Did you check in `/` to see if similar tests already existed there?  I would not be surprised if there were similar ones, but also not surprised if there were not.  :)

File: test/config_selector_test.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
+/*
```
[the remark this answers is not in the document]
Comment (easwars) [reply test/config_selector_test.go]: Done. 

Earlier, I think I had done a `git mv test/resolver_test.go test/config_selector_test.go`. But then, added the new tests to a new file `test/resolver_test.go`. But `git` knew it as a old file. Hence the confusion. Thanks.

[the remark this answers is not in the document]
Comment (easwars) [reply test/config_selector_test.go]: So, I had to put the tests from `resolver_conn_wrapper_test.go` and the new tests into `resolver_update_test.go`.

**Comment by easwars** [comment]
> Nice, thanks for adding the new test cases. Did you check in `/` to see if similar tests already existed there? I would not be surprised if there were similar ones, but also not surprised if there were not. :)

I had done from basic grepping, and the only tests were the ones in the now deleted `resolver_conn_wrapper_test.go`. They were incomplete.
